compile: slicing an array does not produce a nil check

This behavior is easily demonstrated here:
```
package main

func main() {
        var x *[4]uint32
        println(x[:][0])
}
```

Expected behavior (`go run`):
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x45dc9f]

goroutine 1 [running]:
main.main()
        /home/niaow/bad.go:5 +0x1f
exit status 2
```

Actual behavior (`tinygo run`):
```
0
```

--- a/compiler/compiler.go
+++ b/compiler/compiler.go
@@ -1852,6 +1852,7 @@ func (b *builder) createExpr(expr ssa.Value) (llvm.Value, error) {
 				low,
 			}
 
+			b.createNilCheck(expr.X, value, "slice")
 			b.createSliceBoundsCheck(llvmLen, low, high, max, lowType, highType, maxType)
 
 			// Truncate ints bigger than uintptr. This is after the bounds

